### ISSUE TITLE
Release of trained models 

### ISSUE THREAD

**Issue by sdeconinck** [SonyResearch/SemSim#5]
Hi there,

Are there any plans to release the trained weights of any of the SemSim models used in the paper (e.g., the one from Table 1)?

**Comment by sxzrt** [comment]
Hi there,
Thank you for reaching out! Please visit  ttps://1drv.ms/u/s!An7SUhdafm_7hfwZwACtR15pmWu6Ag?e=6MVeEd to access the trained weights for one of the SemSim models. Let me know if you need any further assistance.h

**Comment by gparrella12** [comment]
Thank you for providing the link to the trained weights for the SemSim model. I would like to know how to utilize the checkpoint to evaluate the similarity between two images using the provided weights. Could you please provide an example or guide on how to perform this task effectively?

Thank you in advance for your assistance.

**Comment by sxzrt** [comment]
You can use the checkpoint and Semsim_train_evaluation.py to extract features from a set of reconstructed datasets.

To extract features from a single image and calculate the distance between pairs of images, you can use the function below. Please add this function to /benchmark/Semsim_train_evaluation.py:

````
def extract_single_feature(image_path, opt):
    setup = inversefed.utils.system_startup()
    defs = inversefed.training_strategy('bi-conservative')
    defs.epochs = opt.epochs

    model = create_model(opt)
    model.fc = torch.nn.Identity()
    model.to(**setup)
    model.eval()

    # load weight
    root = create_checkpoint_dir(opt)
    filename = os.path.join(root, f'{opt.arch}_{opt.epochs}_re_id.pth')
    if not os.path.exists(filename):
        raise FileNotFoundError(f"Model checkpoint not found: {filename}")

    model.load_state_dict(torch.load(filename))

    # processing image
    transform = transforms.Compose([
        transforms.Resize((32, 32)),  
        transforms.ToTensor(),  
        transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])  
    ])

    image = Image.open(image_path).convert('RGB')
    image = transform(image).unsqueeze(0).to(**setup)

    # extract image features.
    with torch.no_grad():
        feature = model(image).cpu().numpy()

    return feature

if __name__ == '__main__':

    feature1 = extract_single_feature('img1.png',opt)
    feature0 = extract_single_feature('img2.png',opt)
    distance = np.linalg.norm(feature1 - feature0)
    print(f"Feature Distance: {distance}")`
